circuits/aggregator/aggregator_test.go

```diff
@@ -1,34 +1,141 @@
 package aggregator
 
 import (
-	"os"
+	"math"
+	"math/big"
 	"testing"
-	"time"
 
 	"github.com/consensys/gnark-crypto/ecc"
+	fr_bw6761 "github.com/consensys/gnark-crypto/ecc/bw6-761/fr"
+	bw6761mimc "github.com/consensys/gnark-crypto/ecc/bw6-761/fr/mimc"
 	"github.com/consensys/gnark/backend"
+	"github.com/consensys/gnark/frontend"
 	stdgroth16 "github.com/consensys/gnark/std/recursion/groth16"
-	"github.com/consensys/gnark/test"
+	gtest "github.com/consensys/gnark/test"
 	qt "github.com/frankban/quicktest"
+	"github.com/vocdoni/arbo"
+	"github.com/vocdoni/vocdoni-z-sandbox/circuits"
+	ballottest "github.com/vocdoni/vocdoni-z-sandbox/circuits/test/ballotproof"
 	"github.com/vocdoni/vocdoni-z-sandbox/util"
 )
 
-func TestAggregatorCircuit(t *testing.T) {
-	if os.Getenv("RUN_CIRCUIT_TESTS") == "" {
-		t.Skip("skipping circuit tests...")
-	}
+type testCheckInputsCircuit struct {
+	InputsHash       frontend.Variable `gnark:",public"`
+	MaxCount         frontend.Variable
+	ForceUniqueness  frontend.Variable
+	MaxValue         frontend.Variable
+	MinValue         frontend.Variable
+	MaxTotalCost     frontend.Variable
+	MinTotalCost     frontend.Variable
+	CostExp          frontend.Variable
+	CostFromWeight   frontend.Variable
+	EncryptionPubKey [2]frontend.Variable
+	ProcessId        frontend.Variable
+	CensusRoot       frontend.Variable
+	Nullifiers       [MaxVotes]frontend.Variable
+	Commitments      [MaxVotes]frontend.Variable
+	Addresses        [MaxVotes]frontend.Variable
+	EncryptedBallots [MaxVotes][MaxFields][2][2]frontend.Variable
+}
+
+func (c *testCheckInputsCircuit) Define(api frontend.API) error {
+	return checkInputs(api, c.InputsHash, c.MaxCount, c.ForceUniqueness,
+		c.MaxValue, c.MinValue, c.MaxTotalCost, c.MinTotalCost, c.CostExp,
+		c.CostFromWeight, c.ProcessId, c.CensusRoot, c.EncryptionPubKey,
+		c.Nullifiers[:], c.Commitments[:], c.Addresses[:], c.EncryptedBallots[:])
+}
+
+func TestCheckInputs(t *testing.T) {
 	c := qt.New(t)
-	// inputs generation
-	now := time.Now()
-	processId := util.RandomBytes(20)
-	_, placeholder, assigments, err := GenInputsForTest(processId, 3)
+
+	processId := arbo.BigToFF(ecc.BW6_761.ScalarField(), new(big.Int).SetBytes(util.RandomBytes(20)))
```

Comment:
I feel we must have BigToFF() as an internal helper, we use it a lot and does not make much sense to have it within the merkle-tree specific implementation.

Reply:
This is a good task for Claude, btw :)